flang/lib/Lower/Bridge.cpp

```diff
@@ -2500,6 +2509,11 @@ class FirConverter : public Fortran::lower::AbstractConverter {
               mlir::emitError(loc, "symbol \"")
                   << toStringRef(sym.name()) << "\" must already be in map";
             return;
+          } else if (isResult) {
+            // Some Fortran results may be passed by argument (e.g. derived
+            // types)
```

Comment:
The calling convention should be sorted out in Tilikum (like boxchar).

Reply:
Mmmh, I think this point deserves a bit of thinking of what would be the exact cost/benefit and design of doing so.

The situation with passed result seems a bit more complex to delay than boxchar (where it is mainly a question of splitting organizing the arguments) because there is the matter of the result variable storage allocation. Especially if we want to allocate the temp on the heap, then the caller will have to clean-up once the var is used. So far (at least in array expr), temp clean-up is something controlled in lowering (with the statement context), so it seems easier to expose the allocation in lowering.

Now, I am not saying there is no way delaying this to Tilikum, I'll think/experiment, but I do not plan to solve this comment as part of this PR. It would deserve its own PR if we go that way.